### DISCUSSION TITLE
markdown-it: are all plugins supported?

### DISCUSSION THREAD

**Discussion by akil-rails** [vuejs/vitepress#1060] [Q&A] [answered]
When reading the documentation, it appears that all markdown-it plugins are supported. But this doesn't seem to be the case.

For example, [markdown-it-codetabs](https://github.com/cncws/markdown-it-codetabs) , generates a broken UI.

Site: https://akil-rails.github.io/vitepress-starter/markdown-it-plugins.html
Code: https://github.com/akil-rails/vitepress-starter

If indeed only a set of plugins are supported out of the box, a section in the documentation that lists the supported markdown plugins be useful for readers.

**Comment by brc-dd** [comment, marked as answer]
Where have you added its styles? From that plugin's readme:

> Import your own styles, you can start from codetabs.scss.

Most of them are supported, but some may have conflicts with our in-built plugins and/or our styles. Some have issues with Vue itself like katex/mathjax ones. These may require additional config to make them work.

**Reply by akil-rails** [reply]
Using the default style (codetabs.scss), doesn't work either. I suppose there is a conflict with the vitepress styles.

Site: https://akil-rails.github.io/vitepress-starter/markdown-it-plugins.html
Code: https://github.com/akil-rails/vitepress-starter

**Reply by brc-dd** [reply]
@akil-rails Like I said they may require some additional config. In this case, you need to change `pre` in that scss file to `div[class*='language-']`. Here is a demo: https://stackblitz.com/edit/vite-6s7rfj?file=docs%2F.vitepress%2Ftheme%2Fcodetabs.css

![image](https://user-images.githubusercontent.com/40380293/180730607-eda681f9-e470-4de5-9148-c986036c2495.png)

**Reply by akil-rails** [reply]
Thank you @brc-dd .
